jobQueue/setup.js

```diff
@@ -18,19 +18,46 @@ const conf = require('../config');
 const featureToggles = require('feature-toggles');
 const urlParser = require('url');
 const kue = require('kue');
+const activityLogUtil = require('../utils/activityLog');
+
 const redisOptions = {
   redis: conf.redis.instanceUrl.queue,
 };
+
+const express = require('express');
+const server = require('http').Server(express());
+const SHUTDOWN_TIMEOUT = 5000;
+const SUCCESSFUL_EXIT = 0;
+
 const redisInfo = urlParser.parse(redisOptions.redis, true);
 if (redisInfo.protocol !== PROTOCOL_PREFIX) {
   redisOptions.redis = 'redis:' + redisOptions.redis;
 }
 
 const jobQueue = kue.createQueue(redisOptions);
 
+process.on('SIGTERM', () => {
+  const start = Date.now();
+  server.close(() => { // Stop http server and stop accepting any new request.
+    jobQueue.shutdown(SHUTDOWN_TIMEOUT, (err) => {
+      if (featureToggles.isFeatureEnabled('enableSigtermActivityLog')) {
+        const status = `Job queue shutdown: ${err || 'OK'}`;
+        const logWrapper = {
+          status,
+          totalTime: `${Date.now() - start}ms`,
+        };
+        activityLogUtil.printActivityLogString(logWrapper, 'sigterm');
+      }
+
+      process.exit(SUCCESSFUL_EXIT);
```

Comment:
Do we really need to explicitly call process.exit here? server.close() says stop accepting new requests, but if we call process.exit immediately, that doesn't actually give any in-flight requests any time to finish!

Reply:
That's is based on Kue [here](https://github.com/Automattic/kue/blob/87d61503d3d9cc024633efc7611bd25551f0f87d/Readme.md#graceful-shutdown) "When all workers tell Kue they are stopped fn is called" and [here one example](https://github.com/Automattic/kue/blob/f0e6c065c70ea81dbe507d099f3e5d76728c31b0/examples/shutdown.js#L27). My understanding is that callback fn will be called after all jobs finish down.

I didnt create a test for that because that's is supposed to be handled by the framework.